bazel/defs.bzl

```diff
@@ -0,0 +1,135 @@
+"""Provider and rules for generating xacro output at build time."""
+
+XacroInfo = provider(
+    "Provider holding the result of a xacro generation step.",
+    fields = ["result", "data"],
+)
+
+def _xacro_impl(ctx):
+    if ctx.outputs.out:
+        out = ctx.outputs.out
+    else:
+        src = ctx.file.src.basename
+        if not src.endswith(".xacro"):
+            fail("xacro_file src should be named *.xacro not {}".format(src))
+        out = ctx.actions.declare_file(src[:-6])
```

Comment:
Derive this from the length of ".xacro" for readability.

Reply:
Done.